controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook.go

```diff
@@ -60,22 +61,46 @@ func (in *KubeadmControlPlane) ValidateCreate() error {
 	return nil
 }
 
+const (
+	spec                 = "spec"
+	kubeadmConfigSpec    = "kubeadmConfigSpec"
+	clusterConfiguration = "clusterConfiguration"
+)
+
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (in *KubeadmControlPlane) ValidateUpdate(old runtime.Object) error {
+	// add a * to indicate everything beneath is ok.
+	// For example, {"spec", "*"} will allow any path under "spec" to change, such as spec.infrastructureTemplate.name
+	allowedPaths := [][]string{
+		{"metadata", "*"},
+		{spec, kubeadmConfigSpec, clusterConfiguration, "etcd", "local", "imageRepository"},
+		{spec, kubeadmConfigSpec, clusterConfiguration, "etcd", "local", "imageTag"},
+		{spec, "infrastructureTemplate", "name"},
+		{spec, "replicas"},
+	}
+
 	allErrs := in.validateCommon()
 
 	prev := old.(*KubeadmControlPlane)
-	if !reflect.DeepEqual(in.Spec.KubeadmConfigSpec, prev.Spec.KubeadmConfigSpec) {
-		allErrs = append(
-			allErrs,
-			field.Forbidden(
-				field.NewPath("spec", "kubeadmConfigSpec"),
-				"cannot be modified",
-			),
-		)
-	}
+	patch := client.MergeFrom(prev)
```

Comment:
I _think_ this is pulling in spdystream and goproxy, maybe. This code is pretty straight-forward if you wanted to try to eliminate those dependencies - here's what `client.MergeFrom` and `patch.Data` do:

```go
originalJSON, err := json.Marshal(s.from)
	if err != nil {
		return nil, err
	}

	modifiedJSON, err := json.Marshal(obj)
	if err != nil {
		return nil, err
	}

	return jsonpatch.CreateMergePatch(originalJSON, modifiedJSON)
```

Reply:
for sure, always happy to remove dependencies